include/upipe-freetype/upipe_freetype.h

```diff
@@ -4,8 +4,104 @@
 extern "C" {
 #endif
 
+#include <upipe/ubase.h>
+#include <upipe/upipe.h>
+
 #define UPIPE_FREETYPE_SIGNATURE UBASE_FOURCC('f','r','t','2')
 
+/** @This enumarates the freetype pipe commands. */
+enum upipe_freetype_command {
+    /** sentinel */
+    UPIPE_FREETYPE_SENTINEL = UPIPE_CONTROL_LOCAL,
+    /** get the string bonding box (const char *, struct upipe_freetype_bbox *)
+     */
+    UPIPE_FREETYPE_GET_BBOX,
+    /** get the rendered string width (const char *, uint64_t *) */
+    UPIPE_FREETYPE_GET_WIDTH,
+    /** set the freetype pixel size (unsigned) */
+    UPIPE_FREETYPE_SET_PIXEL_SIZE,
+    /** set the baseline position in the buffer */
+    UPIPE_FREETYPE_SET_BASELINE,
+};
+
+/** @This describes a string bonding box. */
+struct upipe_freetype_bbox {
+    struct {
+        /** maximum x value */
+        int x;
+        /** maximum y value */
+        int y;
+    } max;
+    struct {
+        /** minimum x value */
+        int x;
+        /** minimum y value */
+        int y;
+    } min;
+};
+
+/** @This gets the bounding box for a string.
+ *
+ * @param upipe description structure of the pipe
+ * @param str a string
+ * @param height_p filled with the height of the renderered string
+ * @param width_p filled with the width of the rendered string
+ * @return an error code
+ */
```

Comment:
Params do not match the function prototype.

Reply:
Fixed